message_app/static/js/conversations/main.js

```diff
@@ -0,0 +1,79 @@
+// ===== Conversation =====
+const CONVERSATION_PANEL_HTML = document.querySelector("#conversation-container");
+
+class ConversationModel {
+
+    constructor(participantId, lastMessage) {
+        this.participantId = participantId;
+        this.lastSenderId = lastMessage.senderId;
+        this.timestamp = lastMessage.timestamp;
+        this.messageContent = lastMessage.content;
+        this.HTMLElement = new ConversationHTMLElement(this);
+    }
+
+    show() {
+        this.HTMLElement.show();
+    }
+}
+
+class ConversationHTMLElement {
```

Comment:
Does this class generate all the HTML for us?

Reply:
It generates the HTML element for a conversation on the conversation listing panel.